js/customize-object-selector-component.js

```diff
@@ -366,6 +366,47 @@ wp.customize.ObjectSelectorComponent = (function( api, $ ) {
 		},
 
 		/**
+		 * Setup links for editing objects in select2.
+		 *
+		 * @returns {void}
+		 */
+		setupEditLinks: function setupEditLinks() {
+			var component = this;
+
+			// Set up the add new post buttons
+			component.container.on( 'click', '.select2-selection__choice__edit', function() {
+				var ensuredPromise, returnPromise, postId, $elm = $( this );
```

Comment:
The body of this function can now be replaced with a call to `api.Posts.startEditPostFlow()`. See https://github.com/xwp/wp-customize-posts/blob/33840418da9dcd7c55d15061de5fe49f85d9058f/js/customize-posts.js#L589-L598